### ISSUE TITLE
Add options to customize which columns show up in tables (Flows, Executions, etc)

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#6237]
### Feature description

Before 0.20, users could select which columns show up within filter. With the new filter functionality, selecting displayable columns via filter seems like a bad UX - @Nico-Kestra we need to figure out some alternative way to handle that kind of customization: possibly near the table header itself.

![image](https://github.com/user-attachments/assets/1f1213a3-9dc3-4245-b591-8df0acd81a60)

**Comment by Nico-Kestra** [comment]
We had planned for that, no?
<img width="1746" alt="Capture d’écran 2024-12-02 à 14 54 29" src="https://github.com/user-attachments/assets/595a7b6e-dd5d-4658-a31e-1b297d07d112">

**Comment by MilosPaunovic** [comment]
Yes, but it is not part of the same structure - we use filters for data filtering on backend, that option is related only to local view of the table columns, and is more of a table thing then filter one.

**Comment by anna-geller** [comment]
no strong opinions, what @Nico-Kestra showed works for me - @MilosPaunovic please decide or coordinate with Nico during cooldown, maybe schedule a short sync meeting together tho discuss this and other UI issues to which Nico is assigned? 🙏

**Comment by Nico-Kestra** [comment]
<img width="946" alt="Capture d’écran 2024-12-03 à 10 23 19" src="https://github.com/user-attachments/assets/734c51ba-e58b-44b3-acab-bffc30f7a3ae">
Maybe something look like Notion table?

![sort](https://github.com/user-attachments/assets/7dfc0b22-cc97-461e-93da-0977d80a8964)

**Comment by MilosPaunovic** [comment]
This actually looks very nice and useful, it gets my vote.

**Comment by Nico-Kestra** [comment]
<img width="855" alt="Capture d’écran 2024-12-03 à 10 42 53" src="https://github.com/user-attachments/assets/49861e7c-c777-4470-a48c-29fe66fbe9d6">
WDYT?

**Comment by MilosPaunovic** [comment]
Perfect. Only drop the `New property` button, as we can only show a limited list of items, the ones we get from the backend.

**Comment by Nico-Kestra** [comment]
ok, done!

<img width="418" alt="Capture d’écran 2024-12-03 à 11 23 28" src="https://github.com/user-attachments/assets/a78e1e7c-1054-472f-add5-14b939978837">


[Figma](https://www.figma.com/design/ew0uXk0NRXJ2NBBJTNe2n1/01_UI?node-id=3860-39246&t=nzh6ymxchj6C7lGT-1)

**Comment by Piyush-r-bhaskar** [comment]
Hi @MilosPaunovic & @anna-geller 

I managed to implement this feature with a custom component for `Properties`. I am currently facing challenges with the implementation alongside the `Filter`. As a temporary solution, I have placed it above the table. Would this approach be acceptable from a product perspective?

***Let me know then I will open a PR for this.*** & _Also, I am yet to do it for `Triggers`._


**Overview of logics implemented for better experience:**

- Added logic to **_show the button to select column only when there is data in the table_**.
- Added logic to **_keep the column selection and hide even on refresh_**.
- Removed top Pagination & added **_Properties_** component **_to be displayed above the table_**.
- **_Show 'Hidden in table' or 'Shown in table' title only if there is any column under_** this.
- Added **_additional logic to hide options container on clicking outside anywhere on screen other than toggle on button_**.
- Added **_logic to search for a column._**
- Translation is also working for column name under container 


https://github.com/user-attachments/assets/fc4ae0a3-74e4-4096-88ce-2eebb109de03

**Comment by anna-geller** [comment]
> As a temporary solution, I have placed it above the table. Would this approach be acceptable from a product perspective?

Sure, it sounds great as a first solution; definitely open a PR; big thanks!

**Comment by Nico-Kestra** [comment]
For the properties button, i would have placed it next to the search bar, wouldn't you?


<img width="1840" alt="Capture d’écran 2025-01-09 à 10 38 24" src="https://github.com/user-attachments/assets/c2f9eb8b-f395-47a7-894a-2a41b9561fe6" />


[Figma](https://www.figma.com/design/ew0uXk0NRXJ2NBBJTNe2n1/01_UI?node-id=3433-34304&t=l9ZHDOYq9B9LK52E-1)

**Comment by Piyush-r-bhaskar** [comment]
Thank you @MilosPaunovic . **New PR**- https://github.com/kestra-io/kestra/pull/6932

**Comment by MilosPaunovic** [comment]
This is quite nice improvement, thanks so much for this @Piyush-r-bhaskar! 🚀
